### PR TITLE
improvement(perf regression pipeline): add base version calculator

### PR DESCRIPTION
**Pull request by fgelcer**
since we now have a performance regression test that does latency test during upgrade nodes operation,
we had the need to add an automatic calculator to get the base version, based on the repository URL passed as a parameter to the test.

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by fgelcer** [comment]
saw it working in [this job](https://jenkins.scylladb.com/job/scylla-staging/job/fabio/job/perf-during-upgrade/job/latency_650gb_during_upgrade/), both for master, and for [master] enterprise.

important to note that for these versions, we already have 1 single base version to test, but for releases, we should have between 2 and 4 base versions, and it won't give them all automatically, because it will demand to completely refactor the pipeline, so the current implementation will select the "last" one in the supported base version...

**Review by temichus — CHANGES_REQUESTED** [review]
@fgelcer Not sure that i understand this PR
in the commit message you write ```we had the need to add an automatic calculator to get the base version, based on the repository URL passed as a parameter to the test.```


but in the code you provide base_versions **manually**, 
then **validate** that they in supportedUpgradeFromVersions
and **take the last one** in not params.scylla_ami_id provided 

this is not an ```automatic calculator to get the base version, based on the repository URL passed as a parameter```

also, I can't find a successful job run where used base_versions param and logik got thought new elif statement:
```
elif [[ ! -z "${supportedVersions}" ]]; then
export SCT_SCYLLA_VERSION=${supportedVersions.last()}
```

**Comment by fgelcer** [comment]
> @fgelcer Not sure that i understand this PR in the commit message you write `we had the need to add an automatic calculator to get the base version, based on the repository URL passed as a parameter to the test.`
> 
> but in the code you provide base_versions **manually**, then **validate** that they in supportedUpgradeFromVersions and **take the last one** in not params.scylla_ami_id provided
> 
> this is not an `automatic calculator to get the base version, based on the repository URL passed as a parameter`
> 
> also, I can't find a successful job run where used base_versions param and logik got thought new elif statement:
> 
> ```
> elif [[ ! -z "${supportedVersions}" ]]; then
> export SCT_SCYLLA_VERSION=${supportedVersions.last()}
> ```

yes, it is automatic calculator, exactly as the upgrade tests work... the thing is, it has the ABILITY to also receive manual input to base version, but if not provided, it will be calculated, and proven it works in [this job](https://jenkins.scylladb.com/job/scylla-staging/job/fabio/job/perf-during-upgrade/job/latency_650gb_during_upgrade/)

**Comment by amoskong** [comment]
> @fgelcer Not sure that i understand this PR in the commit message you write `we had the need to add an automatic calculator to get the base version, based on the repository URL passed as a parameter to the test.`
> 
> but in the code you provide base_versions **manually**, then **validate** that they in supportedUpgradeFromVersions and **take the last one** in not params.scylla_ami_id provided

The `base_version` parameter isn't a particular scylla release version.
The particular version needs to be parsed and detected from the repo content (packages list).

You can find the implementation in:
- ./vars/supportedUpgradeFromVersions.groovy
- sct.py: get_scylla_base_versions()
- utils/get_supported_scylla_base_versions.py

```
# We support to migrate from specific OSS version to enterprise
supported_src_oss = {'2021.1': '4.3', '2022.1': '5.0', '2022.2': '5.1', '2023.1': '5.2', '2023.2': '5.3'}
# If new support distro shared repo with others, we need to assign the start support versions. eg: centos8
start_support_versions = {'centos-8': {'scylla': '4.1', 'enterprise': '2021.1'}}
start_support_backend = {'azure': {'scylla': '5.2', 'enterprise': '2023.1'}}
```

The init commit is https://github.com/scylladb/scylla-cluster-tests/commit/c65063144e5634b2b362514007b6a6f2783b543a
 
> this is not an `automatic calculator to get the base version, based on the repository URL passed as a parameter`

**Comment by fgelcer** [comment]
ping @fruch @temichus @aleksbykov @roydahan

File: vars/perfRegressionParallelPipeline.groovy [hunk 1]
```diff
@@ -125,6 +130,15 @@ def call(Map pipelineParams) {
                                     dir('scylla-cluster-tests') {
                                         checkout scm
                                         (testDuration, testRunTimeout, runnerTimeout, collectLogsTimeout, resourceCleanupTimeout) = getJobTimeouts(params, builder.region)
+                                        base_versions_list = params.base_versions.contains('.') ? params.base_versions.split('\\,') : []
+                                        // NOTE: master and enterprise will have 1 single version, but for releases,
+                                        // we will choose automatically only the last one
+                                        supportedVersions = supportedUpgradeFromVersions(
+                                            base_versions_list,
+                                            'ubuntu-focal',
```
Comment (fruch) [inline vars/perfRegressionParallelPipeline.groovy:138]: Why hard-coding this one ?, and not putting it on the pipelines params ?

Comment (fgelcer) [reply vars/perfRegressionParallelPipeline.groovy:138]: first of all, to reduce complexity (we don't need more complexity for now)...
secondly, we could add another parameter that defaults to `ubuntu-focal`, but our images are ubuntu based, and i don't remember a use case where we used a different one for perf tests

Comment (fruch) [reply vars/perfRegressionParallelPipeline.groovy:138]: I still don't lie the idea of having something like that hardcoded in the middle of the pipeline, but let's leave it for now

**Pull request merged**